java-frontend/src/test/java/org/sonar/java/cfg/CFGDebug.java

```diff
@@ -17,20 +17,104 @@
  * along with this program; if not, write to the Free Software Foundation,
  * Inc., 51 Franklin Street, Fifth Floor, Boston, MA  02110-1301, USA.
  */
-package org.sonar.java.model;
+package org.sonar.java.cfg;
 
-import org.sonar.java.cfg.CFG;
 import org.sonar.java.cfg.CFG.Block;
+import org.sonar.java.model.SyntaxTreeDebug;
 import org.sonar.plugins.java.api.tree.Tree;
 import org.sonar.plugins.java.api.tree.Tree.Kind;
 
+import java.util.List;
+
 public class CFGDebug {
 
+  private static final String DOT_PADDING = "  ";
+  private static final String DOT_NEW_LINE = "\\n";
+
   private static final int MAX_KINDNAME = Kind.UNSIGNED_RIGHT_SHIFT_ASSIGNMENT.name().length() + 5;
 
   private CFGDebug() {
   }
 
+  public static String toDot(CFG cfg) {
+    StringBuilder sb = new StringBuilder();
```

Comment:
You would probably be better off using StringFormat rather than a Striing builder for most of your cases.